### PR TITLE
refactor(api): remove groups and scopes from worker consumer

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Review by sguiheux — APPROVED** [review]

File: engine/api/worker/registration.go [hunk 1]
```diff
@@ -58,42 +58,69 @@ type TakeForm struct {
...
+func RegisterWorker(ctx context.Context, db gorpmapper.SqlExecutorWithTx, store cache.Store,
+	spawnArgs hatchery.SpawnArgumentsJWT, hatcheryService sdk.Service, hatcheryConsumer *sdk.AuthConsumer, workerConsumer *sdk.AuthConsumer,
+	registrationForm sdk.WorkerRegistrationForm, runNodeJob *sdk.WorkflowNodeJobRun) (*sdk.Worker, error) {
+	if err := spawnArgs.Validate(); err != nil {
+		return nil, err
 	}
 
 	var model *sdk.Model
-	if spawnArgs.Model != nil {
+	if spawnArgs.Model.ID > 0 {
```
[the remark this answers is not in the document]
Comment (richardlt) [reply engine/api/worker/registration.go:69]: Model can't be nil now, so we check the value of the ID instead.

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16919.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16926.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16926.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: engine/api/worker/registration.go [hunk 2]
```diff
@@ -58,42 +58,69 @@ type TakeForm struct {
...
 		var err error
 		model, err = workermodel.LoadByID(ctx, db, spawnArgs.Model.ID, workermodel.LoadOptions.Default)
 		if err != nil {
 			return nil, err
 		}
 	}
+	if spawnArgs.RegisterOnly && model == nil {
```
[the remark this answers is not in the document]
Comment (richardlt) [reply engine/api/worker/registration.go:76]: We check model only if set, will be ok for local hatchery as no model should be given.

**Pull request merged**